### PR TITLE
update get and edit user

### PR DESCRIPTION
**Pull request by BelehovEgor**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/777

without communication

### CODE REVIEW

File: src/UserService.Business/Commands/User/EditUserCommand.cs [hunk 1]
```diff
@@ -76,38 +198,49 @@ public class EditUserCommand : IEditUserCommand
...
         public OperationResultResponse<bool> Execute(Guid userId, JsonPatchDocument<EditUserRequest> patch)
         {
             var status = OperationResultStatusType.FullSuccess;
 
-            if (!(_userRepository.Get(_httpContext.GetUserId()).IsAdmin ||
+            if (!(_userRepository.Get(_httpContextAccessor.HttpContext.GetUserId()).IsAdmin ||
                 _accessValidator.HasRights(Rights.AddEditRemoveUsers) ||
-                (userId == _httpContext.GetUserId() &&
+                (userId == _httpContextAccessor.HttpContext.GetUserId() &&
                 !(patch.Operations.FirstOrDefault(o => o.path.EndsWith(nameof(EditUserRequest.Rate), StringComparison.OrdinalIgnoreCase)) != null))))
```
Comment (TIove) [inline src/UserService.Business/Commands/User/EditUserCommand.cs]: without '!' and set ==

Comment (BelehovEgor) [reply src/UserService.Business/Commands/User/EditUserCommand.cs]: fixed

**Review by MariyaSin — APPROVED** [review]

**Review by TIove — APPROVED** [review]

File: src/UserService.Business/Commands/User/EditUserCommand.cs [hunk 2]
```diff
@@ -25,12 +28,131 @@ namespace LT.DigitalOffice.UserService.Business
...
+            {
+                Response<IOperationResult<bool>> response = _rcDepartment.GetResponse<IOperationResult<bool>>(
+                    IChangeUserDepartmentRequest.CreateObj(userId, departmentId)).Result;
+
+                if (response.Message.IsSuccess && response.Message.Body)
+                {
+                    return true;
+                }
+
+                _logger.LogWarning(logMessage, userId, departmentId);
```
Comment (djino97) [inline src/UserService.Business/Commands/User/EditUserCommand.cs:59]: "Reason: {Errors}" from response.Message.Errors

File: src/UserService.Business/Commands/User/EditUserCommand.cs [hunk 3]
```diff
@@ -25,12 +28,131 @@ namespace LT.DigitalOffice.UserService.Business
...
+            {
+                Response<IOperationResult<bool>> response = _rcPosition.GetResponse<IOperationResult<bool>>(
+                    IChangeUserPositionRequest.CreateObj(userId, positionId)).Result;
+
+                if (response.Message.IsSuccess && response.Message.Body)
+                {
+                    return true;
+                }
+
+                _logger.LogWarning(logMessage, positionId, userId);
```
Comment (djino97) [inline src/UserService.Business/Commands/User/EditUserCommand.cs:86]: "Reason: {Errors}" from response.Message.Errors

**Review by djino97 — APPROVED** [review]

File: src/UserService.Business/Commands/User/EditUserCommand.cs (hunk 2 above)
Comment (BelehovEgor) [reply src/UserService.Business/Commands/User/EditUserCommand.cs:59]: fixed

File: src/UserService.Business/Commands/User/EditUserCommand.cs (hunk 3 above)
Comment (BelehovEgor) [reply src/UserService.Business/Commands/User/EditUserCommand.cs:86]: fixed

**Pull request merged**